dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt

```diff
@@ -158,6 +160,20 @@ internal object CoreFeature {
         )
     }
 
+    fun drainExecutors() {
+        val tasks = ArrayList<Runnable>()
+        (persistenceExecutorService as? ThreadPoolExecutor)
+            ?.queue
+            ?.drainTo(tasks)
```

Comment:
I just checked implementation details of `ThreadPoolExecutor` and since `shutdownNow` returns tasks which were in the queue, after this call `queue` is actually empty, because it is drained during this call.

http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/concurrent/ThreadPoolExecutor.java#l1414

Reply:
Yes...I see but I am not sure if I want to shutdown the executors here ...as this is a flush method and the `Datadog#stop` method is already handling this. I am thinking that we might need at some point just to flush the data but let the executors work after, if we shut them down here we will no longer be able to use them